src/components/contexts/fluxes-waves/FluxesWavesProvider.js

```diff
@@ -0,0 +1,227 @@
+
+  const cloudToGroundWaveBeginsX = sunCenterX;
+  const cloudToGroundWaveBeginsY = cloudBottomY;
+  const cloudToGroundWaveEndsY = groundBeginsY;
+  const cloudToGroundWaveStartsAfterInterval = sunToCloudWaveReachesEnd;
+  const cloudToGroundWaveReachesEnd =
+    sunToCloudWaveReachesEnd +
+    Math.abs(cloudToGroundWaveEndsY - cloudToGroundWaveBeginsY) /
+      Y_SHIFT_PER_INTERVAL;
+
+  const cloudToAtmosphereWaveBeginsX =
+    firstCloudCentralCircleX - firstCloudCentralCircleRadiusX;
+  const cloudToAtmosphereWaveBeginsY = firstCloudCentralCircleY;
+  const cloudToAtmosphereWaveEndsY = sunBottomY;
+  const cloudToAtmosphereWaveStartsAfterInterval = sunToCloudWaveReachesEnd;
+
+  const groundToAtmosphereWaveBeginsX =
+    firstIceCapRowBeginsX + iceCapBaseWidth / 2;
+  const groundToAtmosphereWaveBeginsY = firstIceCapRowBeginsY - iceCapHeight;
+  const groundToAtmosphereWaveEndsY =
+    firstIceCapRowBeginsY - iceCapHeight - skyHeight;
+  const groundToAtmosphereWaveStartsAfterInterval = cloudToGroundWaveReachesEnd;
+
+  // earth wave dimensions, positions, starting/ending intervals
+  const groundToSkyWaveBeginsX =
+    GROUND_TO_SKY_EARTH_FLUX_ADJUSTMENT * (stageWidth - seaWidth);
+  const groundToSkyWaveBeginsY = groundBeginsY;
+  const groundToSkyWaveEndsY = groundBeginsY - skyHeight / 2;
+  const groundToSkyWaveStartsAfterInterval = cloudToGroundWaveReachesEnd;
+  const groundToSkyWaveReachesEnd =
+    cloudToGroundWaveReachesEnd +
+    Math.abs(groundToSkyWaveEndsY - groundToSkyWaveBeginsY) /
+      Y_SHIFT_PER_INTERVAL;
+
+  const skyToGroundWaveBeginsX =
+    SKY_TO_GROUND_EARTH_FLUX_ADJUSTMENT * (stageWidth - seaWidth);
+  const skyToGroundWaveBeginsY = cloudBottomY;
+  const skyToGroundWaveEndsY = groundBeginsY;
+  const skyToGroundWaveStartsAfterInterval = groundToSkyWaveReachesEnd;
+
+  const skyToAtmosphereWaveBeginsX =
+    SKY_TO_ATMOSPHERE_EARTH_FLUX_ADJUSTMENT * (stageWidth - seaWidth);
+  const skyToAtmosphereWaveBeginsY = firstCloudCentralCircleY;
+  const skyToAtmosphereWaveEndsY = sunBottomY;
+  const skyToAtmosphereWaveStartsAfterInterval = groundToSkyWaveReachesEnd;
+
+  return (
+    <FluxesWavesContext.Provider
+      value={{
+        sunToCloudFluxBeginsX,
+        sunToCloudFluxBeginsY,
+        sunToCloudFluxHeight,
+        sunToCloudFluxStartsAfterInterval,
+        cloudToGroundFluxBeginsX,
+        cloudToGroundFluxBeginsY,
+        cloudToGroundFluxHeight,
+        cloudToGroundFluxStartsAfterInterval,
+        cloudToAtmosphereFluxBeginsX,
+        cloudToAtmosphereFluxBeginsY,
+        cloudToAtmosphereFluxHeight,
+        cloudToAtmosphereFluxStartsAfterInterval,
+        groundToAtmosphereFluxBeginsX,
+        groundToAtmosphereFluxBeginsY,
+        groundToAtmosphereFluxHeight,
+        groundToAtmosphereFluxStartsAfterInterval,
+        groundToSkyFluxBeginsX,
+        groundToSkyFluxBeginsY,
+        groundToSkyFluxHeight,
+        groundToSkyFluxStartsAfterInterval,
+        skyToGroundFluxBeginsX,
```

Comment:
I think it's a bit too many variables. Wouldn't it be possible to group them? Like 
```
{ 
 skyToGround: {
   x, y, height
},
skyToAthmosphere: {
  height, startsAfterInterval, y
}
}
```

Reply:
Good idea